source/extensions/filters/network/thrift_proxy/metadata.h

```diff
@@ -80,10 +109,27 @@ class MessageMetadata {
   absl::optional<AppExceptionType> app_ex_type_;
   absl::optional<std::string> app_ex_msg_;
   bool protocol_upgrade_message_{false};
+  SpanList spans_;
+  absl::optional<int64_t> trace_id_;
+  absl::optional<int64_t> trace_id_high_;
+  absl::optional<int64_t> span_id_;
+  absl::optional<int64_t> parent_span_id_;
+  absl::optional<int64_t> flags_;
+  absl::optional<bool> sampled_;
 };
 
 typedef std::shared_ptr<MessageMetadata> MessageMetadataSharedPtr;
 
+/**
+ * Constant Thrift headers. All lower case.
+ */
+class HeaderValues {
+public:
+  const Http::LowerCaseString ClientId{":client-id"};
+  const Http::LowerCaseString Dest{":dest"};
```

Comment:
should we document these (special?) headers somewhere else too?

Reply:
Sure.